lib/galaxy/web/framework/middleware/remoteuser.py

```diff
@@ -143,6 +145,10 @@ def __call__( self, environ, start_response ):
             # The API handles its own authentication via keys
             return self.app( environ, start_response )
         else:
+            log.debug("Unable to identify user.  %s not found" % self.remote_user_header)
+            for key in environ.keys():
+                log.debug("%s = %s" % (key, environ.get(key)))
```

Comment:
For log, it is best to not do the string substitution manually (with %), but to provide it as bare arguments to the log.xyz function: ```log.debug( "%s = %s", k, v )```